server/app/views/questiontypes/ApplicantQuestionRendererImpl.java

```diff
@@ -36,10 +37,19 @@ protected enum InputFieldType {
 
   protected final ApplicantQuestion question;
   private final InputFieldType inputFieldType;
+  // HTML id tags for various elements within this question.
+  private final String id;
+  private final String descriptionId;
+  private final String requiredErrorId;
+  private final String validationErrorId;
 
   ApplicantQuestionRendererImpl(ApplicantQuestion question, InputFieldType inputFieldType) {
     this.question = checkNotNull(question);
     this.inputFieldType = checkNotNull(inputFieldType);
+    id = question.getContextualizedPath().toString();
```

Comment:
suggestion (blocking): can we convert this to a value that doesn't contain brackets and periods? according to the note [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id) it is a best practice for IDs to only use letters, numbers, `-`, and `_`

Reply:
Yes, that sounds good! This is actually existing behavior (I just moved it from line 137 to here), so I think it doesn't need to be blocking (unless you have concerns about this PR adding more IDs that don't follow that guideline).

I filed https://github.com/civiform/civiform/issues/3356 and can send out a separate PR for that fix.

Reply:
Actually based on Brett's suggestion in the bug to use a random ID (which sounds good to me), I'll just update here since that's a one line change.